admin-dev/themes/new-theme/js/components/country-state-selection-toggler.js

```diff
@@ -52,47 +52,52 @@ export default class CountryStateSelectionToggler {
     this.$countryStateSelector = $(countryStateSelector);
     this.$countryInput = $(countryInputSelector);
 
-    this.$countryInput.on('change', () => this._toggle());
+    this.$countryInput.on('change', () => this._change());
 
     // toggle on page load
-    this._toggle(true);
+    this._toggle();
 
     return {};
   }
 
   /**
-   * Toggles State selection
+   * Change State selection
    *
    * @private
    */
-  _toggle(isFirstToggle = false) {
+  _change() {
+    const countryId = this.$countryInput.val();
+    if (countryId === '') {
+      return;
+    }
     $.ajax({
       url: this.$countryInput.data('states-url'),
       method: 'GET',
       dataType: 'json',
       data: {
-        id_country: this.$countryInput.val(),
-      }
+        id_country: countryId,
+      },
     }).then((response) => {
-      if (response.states.length === 0) {
-        this.$stateSelectionBlock.fadeOut();
+      this.$countryStateSelector.empty();
 
-        return;
-      }
+      Object.keys(response.states).forEach((value) => {
+        this.$countryStateSelector.append($('<option></option>').attr('value', response.states[value]).text(value));
+      });
```

Comment:
```suggestion
      Object.entries(response.states).forEach((key, value) => {
      }
        this.$countryStateSelector.append($('<option></option>').attr('value', key).text(value));
      });
```

Reply:
This won't work because in the data I get, the key is the state name and the value is the state id, so if I use `key` I'll have the index of the array instead of the state id.

Reply:
I'm not sur that's the best solution, `key` will be the index of the array, not the actual array key that I need. Also, `value` will be an array and I'll have to use `value[0]` to get the actual value and `value[1]` to get the array key. I think we should keep the initial proposition, wdyt?